### PR TITLE
feat: change @vue/compiler-sfc to an optional dependency

### LINKED ISSUE #188: @vue/compiler-sfc should be an optionalDependency instead of a peerDependency
**Describe the bug**

This PR https://github.com/trivago/prettier-plugin-sort-imports/commit/157c7f9bf3b08e8b8c69320379a782236ffaf79b made it so v4.0.0 now has `@vue/compiler-sfc` as a peer dependency — which package managers will complain about.

Peer dependencies are supposed to be required in order for the package to work correctly. Instead, `@vue/compiler-sfc` should be marked as an optional dependency.

**Your Environment**
 * **Prettier version**:  2.8.0
 * **node version** 16.18.0
 * **package manager**: pnpm@7..17.0
 * **IDE**: CLI

**Error log**

```
WARN  Issues with peer dependencies found.
└─┬ @trivago/prettier-plugin-sort-imports 4.0.0
  └── ✕ missing peer @vue/compiler-sfc@3.x
Peer dependencies that should be installed:
  @vue/compiler-sfc@3.x  
```

**Contribute to @trivago/prettier-plugin-sort-imports**

- [x] I'm willing to fix this bug 🥇

### PR DESCRIPTION
**Pull request by transitive-bullshit**
Fixes #188

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,9 @@
     "typescript": "4.2.3"
   },
   "peerDependencies": {
-    "@vue/compiler-sfc": "3.x",
     "prettier": "2.x"
+  },
+  "optionalDependencies": {
+    "@vue/compiler-sfc": "3.x"
```
Comment (IanVS) [inline package.json:59]: Optional dependencies are always attempted to be installed by the package manager, but the installation will not fail if it does not complete.  This is intended to be used for packages that can only run in one architecture, for instance. 

The correct approach here is to use `peerDependenciesMeta` like this: https://github.com/IanVS/prettier-plugin-sort-imports/pull/45/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R72-R75

Comment (transitive-bullshit) [reply package.json:57]: Good call; you're definitely right. Just updated the PR w/ this change.

**Comment by transitive-bullshit** [comment]
@byara @trivago polite nudge 😄

**Review by byara — APPROVED** [review]
LGTM 🚀

**Comment by ayusharma** [comment]
Released in [v4.1.0](https://github.com/trivago/prettier-plugin-sort-imports/releases/tag/v4.1.0). I appreciate your patience. ❤️

**Pull request merged**